packages/qwik-city/src/buildtime/vite/plugin.ts

```diff
@@ -186,8 +186,32 @@ function qwikCityPlugin(userOpts?: QwikCityVitePluginOptions): any {
           }
 
           if (isSwRegister) {
+            // Detect if the developer added custom code in their service-worker.ts file
+            let didNotAddCustomCode = false;
+            const clientOutDir = qwikPlugin!.api.getClientOutDir();
+            if (clientOutDir) {
+              const basePathRelDir = api.getBasePathname().replace(/^\/|\/$/, '');
+              const clientOutBaseDir = join(clientOutDir, basePathRelDir);
+              for (const swEntry of ctx.serviceWorkers) {
+                try {
+                  const swClientDistPath = join(clientOutBaseDir, swEntry.chunkFileName);
+                  const swCode = await fs.promises.readFile(swClientDistPath, 'utf-8');
+                  const swCodeTrimmed = swCode.replace(/\s+/g, '');
+                  if (
+                    swCodeTrimmed ===
+                      'addEventListener("install",()=>self.skipWaiting());addEventListener("activate",()=>self.clients.claim());' ||
```

Comment:
oh I see, the problem is that the own service worker is also in there? Can't we change the buildcontext so that doesn't happen?
It seems very brittle to rely on a specific output

Reply:
The problem is that we always register once service-worker.js for everything. So we must look at the code that is in there. I can try to move the code to the build context yeah although we should depracate this service-worker.ts API imo so not that big of a deal if the code isn't too pretty.